core/src/main/java/io/aiven/klaw/config/CacheConfig.java

```diff
@@ -36,4 +55,39 @@ public CacheManager cacheManager() {
     cacheManager.setCaches(Arrays.asList(userCache, teamsCache));
     return cacheManager;
   }
+
+  @Bean
+  public CacheService<Env> kafkaEnvListPerTenant() {
+    return new CacheService<>(CacheConstants.ENVIRONMENT_PATH, HARestMessagingService);
+  }
+
+  @Bean
+  public CacheService<Env> schemaRegEnvListPerTenant() {
+    return new CacheService<>(CacheConstants.ENVIRONMENT_PATH, HARestMessagingService);
+  }
+
+  @Bean
+  public CacheService<Env> kafkaConnectEnvListPerTenant() {
+    return new CacheService<>(CacheConstants.ENVIRONMENT_PATH, HARestMessagingService);
+  }
+
+  @Bean
+  public CacheService<Env> allEnvListPerTenant() {
+    return new CacheService<>(CacheConstants.ENVIRONMENT_PATH, HARestMessagingService);
+  }
+
+  @Bean
+  @ConditionalOnProperty(name = "klaw.login.authentication.type", havingValue = "ad")
```

Comment:
```suggestion
  @ConditionalOnProperty(name = "klaw.login.authentication.type", havingValue = "db")
```
I think this should be db , for inmemory

Reply:
So for Cache Config it loads the App2App user inmemory only when AD is setup everything else is going through AD.

When we use db setup it is actually configured with the other users in user setup.